spec/furniture/marketplace/finalize_order_payment_job_spec.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Marketplace::FinalizeOrderPaymentJob, type: :job do
+  describe "#perform" do
+    subject(:action) { described_class.perform_now(order.id, "123") }
+
+    let!(:order) { create(:marketplace_order) }
+    let(:mailer_double) { instance_double(ActionMailer::MessageDelivery, deliver_later: nil) }
+
+    before do
+      allow(Marketplace::SplitJob).to receive(:perform_later)
+      allow(Marketplace::Order::ReceivedMailer).to receive(:notification).and_return(mailer_double)
+      allow(Marketplace::Order::PlacedMailer).to receive(:notification).and_return(mailer_double)
+      allow(mailer_double).to receive(:deliver_later)
+    end
+
+    it "create order events" do
+      expect { action }.to change { order.events.reload.size }.from(0).to(3)
+    end
+
+    it "call jobs" do
+      action
+      expect(Marketplace::SplitJob).to have_received(:perform_later)
+      expect(mailer_double).to have_received(:deliver_later).exactly(2).times
```

Comment:
`rspec-rails` has [`have_enqueued_mail`](https://www.rubydoc.info/gems/rspec-rails/RSpec%2FRails%2FMatchers:have_enqueued_mail) and [`have_enqueued_job`](https://www.rubydoc.info/gems/rspec-rails/RSpec%2FRails%2FMatchers%3Ahave_enqueued_job) matchers that will allow you to make these assertions without stubbing.

Reply:
I think an alternative solution might be to use the after_commit_everywhere gem.
![image](https://github.com/user-attachments/assets/8302403a-6ba1-42b4-86b3-3f10e1d941ad)

Reply:
I think for now the transaction should be sufficient; since any failing step will revert the transaction and rollback any database changes. We may want to get more thoughtful if things start going weird; but I am OK with a basic transaction / rollback for now.

Reply:
(That said, I really like how the `after_commit_anywhere` gem keeps the `after_commit` to this particular place, rather than relying on a global `after_commit` that checks for a status change!)